handlers.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type DeviceHandlers struct {
+	service *Service
+}
+
+func (h *DeviceHandlers) devicesHandler(w http.ResponseWriter, r *http.Request) {
+	var devPayload DevicePayload
+
+	err := json.NewDecoder(r.Body).Decode(&devPayload)
+	if err != nil {
+		fmt.Printf("handlerError: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	device, err := h.service.AddDevice(&devPayload)
+	if err != nil {
+		switch err.(type) {
```

Comment:
I would simplify this switch. In case it was validation error - return it. In every other case just return 500

```suggestion
		switch err.(type) {
		case *ErrDao:
		   w.WriteHeader(http.StatusBadRequest)
		   w.Write(err.Error())  // return the error - let the user know what was the problem with his HTTP request
		   fmt.Println(err.Error())
		default:
		        fmt.Println(err.Error())
			w.WriteHeader(http.StatusInternalServerError)
		}
```